Artifacts/windows-deploymentgroup/InstallAndConfigureDeploymentGroupAgent.ps1

```diff
@@ -0,0 +1,229 @@
+}
+
+function Configure-DeploymentGroupAgent
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]$accountUrl,
+        [Parameter(Mandatory=$true)]$projectName,
+        [Parameter(Mandatory=$true)]$deploymentGroupName,
+        [Parameter(Mandatory=$true)]$personalAccessToken,
+        [Parameter(Mandatory=$false)][String] [AllowEmptyString()]$agentName,
+        [Parameter(Mandatory=$true)]$agentInstallPath,
+        [Parameter(Mandatory=$false)][String] [AllowEmptyString()]$deploymentAgentTags
+        )
+
+    If ([string]::IsNullOrEmpty($agentName))
+    {
+        $agentName = "$env:COMPUTERNAME"
+    }
+    Write-Host "Configuring Agent: $agentName"
+
+    pushd $agentInstallPath
+    if ([string]::IsNullOrEmpty($deploymentAgentTags)) 
+    {
+        .\config.cmd --deploymentgroup --agent $agentName --runasservice --work '_work' --url $accountUrl --projectname $projectName --deploymentgroupname $deploymentGroupName --auth PAT --token $personalAccessToken --unattended 
+    } else {
+        .\config.cmd --deploymentgroup --agent $agentName --runasservice --work '_work' --url $accountUrl --projectname $projectName --deploymentgroupname $deploymentGroupName --auth PAT --token $personalAccessToken --adddeploymentgrouptags --deploymentgrouptags $deploymentAgentTags --unattended 
+    }
+    if (! $?) 
+    {
+        Write-Host "Deployment agent configuration failed."
+        Exit 1
+    }
+    popd
+}
+
+function Remove-InstallFiles
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]$agentInstallPath,
+        [Parameter(Mandatory=$true)]$agentZipFile
+        )
+
+    pushd $agentInstallPath
+    $agentZip="$agentInstallPath\$agentZipFile";
+    Remove-Item $agentZip;
+    popd
+}
+
+
+function Install-DeploymentGroupAgent 
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]$accountUrl,
+        [Parameter(Mandatory=$true)]$projectName,
+        [Parameter(Mandatory=$true)]$deploymentGroupName,
+        [Parameter(Mandatory=$true)]$personalAccessToken,
+        [Parameter(Mandatory=$false)] [String] [AllowEmptyString()] $deploymentAgentTags,
+        [Parameter(Mandatory=$false)] [String] [AllowEmptyString()] $agentInstallPath,
+        [Parameter(Mandatory=$false)] [String] [AllowEmptyString()] $agentName
+        )
+
+    $ErrorActionPreference="Stop";
+
+    $agentZipFile = "agent.zip"
+
+    try 
+    {
+
+        #Initial checks
+        Test-InstallPrerequisites
+    
+        #Create agent directory
+        $agentInstallPath = New-AgentPath $agentInstallPath
+    
+        #Download Agent
+        Download-DeploymentGroupAgent $accountUrl $personalAccessToken $agentInstallPath $agentZipFile
+
+        #Configure Agent
+        Configure-DeploymentGroupAgent $accountUrl $projectName $deploymentGroupName $personalAccessToken $agentName $agentInstallPath $deploymentAgentTags  
+
+        #Cleanup
+        Remove-InstallFiles $agentInstallPath $agentZipFile
```

Comment:
Consider doing the cleanup in a finally block, so that even when there is an error during configuration step, the cleanup gets executed.